### ISSUE TITLE
vit imagenet training precision issues with amd w6800 gpu + rocm 5.0

### ISSUE THREAD

**Issue by brettkoonce** [google/jax#10761]
per discussion in #2012, breaking this out into a stand-alone issue.

```
pip install "jax[tpu]>=0.2.16" -f https://storage.googleapis.com/jax-releases/libtpu_releases.html

# rollback to old scenic
git clone https://github.com/google-research/scenic.git
cd scenic
git reset --hard 791ffdd5d6e8fcb2b410ad2fb8c89701ce833c67
pip install .
```

The local python install will need a tweak at line 688 in ~/.local/lib/python3.8/site-packages/scenic/model_lib/base_models/model_utils.py

```
import functools
@functools.partial(jax.vmap, in_axes=[0, 0], out_axes=0)
```

After that, set the batch size to 128 in scenic/projects/baselines/configs/imagenet/imagenet_vit_config.py

```
python3 scenic/main.py  --config=scenic/projects/baselines/configs/imagenet/imagenet_vit_config.py --workdir=/OUTPUT_DIR
```

Per discussion in that thread, Nvidia/TPU devices converge to the same result at 100 epochs, whereas the AMD device (a w6800 with rocm 5.0) converges to a lower number (~80% accuracy of the tpu/nvidia baseline).  Per my notes, had similar difference with rocm 4.5 and the same GPU last year.

I am using this specific version of scenic in order to use the tf-dataset 5.1.0 version of imagenet (the dataset handlers were reworked in February).  Going to work on updating to the latest scenic soon.

**Comment by hawkinsp** [comment]
@reza-amd 

On the assumption that this is a numerical bug, I think the obvious debugging step would be to compare the numerical outputs of a single training step and see if they diverge significantly between AMD GPU and a reference implementation. If so, I would try to minimize the computation to a small reproducer.

(I unfortunately cannot help you with this myself, since I don't have access to the necessary hardware.)

**Comment by brettkoonce** [comment]
I have pulled scenic forward to 86a01bc062f63e5495f7e632edc478e3facd497e and am running ablations there.  I cannot use the ROCm jax build script, see #10162.  Will try again with a more recent jax when I get a chance.

**Comment by brettkoonce** [comment]
ROCm 5.2 working, will try again after https://github.com/google-research/scenic/issues/460 is resolved!

**Comment by brettkoonce** [comment]
will try to reproduce with 5.7 or later!
